01-cloudformation/cloudformation1.2.2.yaml

```diff
@@ -0,0 +1,67 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: My most minimal template possible
+
+Parameters:
+    BucketParameter:
+        Description: BucketParameter
+        Type: String
+    UserParameter:
+        Description: UserParameter
+        Type: String
+
+Conditions:
+  PreferredRegion: !Equals 
+    - !Ref AWS::Region
+    - us-east-1
+
+Resources:
+  Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName:
+        'Fn::If':
+          - PreferredRegion
+          - !Join
+            - ''
+            - - !Ref AWS::AccountId
+              - !Ref BucketParameter
+          - !Join
+              - ''
+              - - !Ref AWS::Region
+                - !Ref BucketParameter
+  User:      
+    Type: AWS::IAM::User
+    Properties:
+      Path: "/"
+      UserName: !Ref UserParameter
+      LoginProfile:
+        Password: myP@ssW0rd
+
+  S3ReadOnlyPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties: 
+      Description: A read-only policy for ringer-s3user
+      ManagedPolicyName: ringer-s3policy
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+        - Effect: Allow
+          Action:
+          - s3:Get*
```

Comment:
Try using s3:ListBucket and s3:GetObject to implement principle of least privilege.